src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,110 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int CAPACITY_MULTIPLIER = 2;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getBucketIndex(key);
+        for (Node<K, V> node = table[index]; node != null; node = node.getNext()) {
+            if ((node.getKey() == key) || (key != null && key.equals(node.getKey()))) {
+                node.setValue(value);
+                return;
+            }
+        }
+        addMyNode(key, value, index);
     }
 
     @Override
     public V getValue(K key) {
+        int index = getBucketIndex(key);
+        for (Node<K, V> node = table[index]; node != null; node = node.getNext()) {
+            if (node.getKey() == key || (key != null && key.equals(node.getKey()))) {
+                return node.getValue();
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public void addMyNode(K key, V value, int index) {
+        Node<K, V> firstNode = table[index];
+        table[index] = new Node<>(key, value, firstNode);
+        if (++size > (int) (table.length * DEFAULT_LOAD_FACTOR)) {
+            resize();
+        }
+    }
+
+    public void transfer(Node<K, V>[] newTable) {
+        for (Node<K, V> currentNode : table) {
+            while (currentNode != null) {
+                Node<K, V> nextNode = currentNode.getNext();
+                int index = getNewBucketIndex(currentNode.getKey(), newTable.length);
+                currentNode.setNext(newTable[index]);
+                newTable[index] = currentNode;
+                currentNode = nextNode;
+            }
+        }
+    }
+
+    private int getNewBucketIndex(K key, int newTableLength) {
+        int hash = (key == null) ? 0 : key.hashCode();
+        return (hash & 0x7fffffff) % newTableLength;
```

Comment:
interesting approach. Could you please describe how it works?

Reply:
ChatGPT helps me with this line (hash & 0x7fffffff) % newTableLength;, because sometimes i have negative hash code. The &0x7fffffff operator is used to set the most significant bit to 0, making the hash code always non-negative.